app/controllers/application_controller.rb

```diff
@@ -175,15 +175,20 @@ def posts_from_relation(relation, no_tests: true, with_pagination: true, select:
         post = posts.detect { |p| p.id == view.post_id }
         post && view.read_at < post.tagged_at
       end.map(&:post_id)
+
+      if with_unread
+        @unread_counts = Reply.where(post_id: @unread_ids).joins('INNER JOIN post_views ON replies.post_id = post_views.post_id')
+        @unread_counts = @unread_counts.where('replies.created_at > post_views.read_at').group(:post_id).count
+      end
```

Comment:
Can we add a test for how `posts_from_relation` works with `with_unread: true`, maybe checking that it gets the right number for 2, 1, and 0 unread replies (and skips a post that's fully unread)?